src/applications/common/schemaform/save-in-progress/actions.js

```diff
@@ -196,23 +204,32 @@ export function saveAndRedirectToReturnUrl(...args) {
  *                                version of the form the data was saved with
  *                                is different from the current version.
  */
-export function fetchInProgressForm(formId, migrations, prefill = false, prefillTransformer = null) {
+export function fetchInProgressForm(formId, migrations, prefill = false, prefillTransformer = null, prestartForm = null) {
+
   // TODO: Migrations currently aren’t sent; they’re taken from `form` in the
   //  redux store, but form.migrations doesn’t exist (nor should it, really)
-  return (dispatch, getState) => {
+  return async (dispatch, getState) => {
+    if (prestartForm) {
+      try {
+        await prestartForm();
+      // If prestart error, fail safely
+      } catch (err) {
+        return;
```

Comment:
Thanks @cvalarida, whether to use `try/catch` or `if` here is an important question, feedback on this gating logic is especially welcome. The user will have an alert displayed if the ITF fails (`ITFErrorAlert`) and the user will be directed to the form error page, and an error will also be logged as soon as the ITF fetch fails in `fetchITF`.

Reply:
Does `prestartForm` throwing an error put the application in an invalid state? I'm not sure what it does.

Reply:
Throwing an error here routes the user to the form error page. Would it make more sense to set this up as an if else statement instead?

Reply:
This now uses an if/else statement rather than a try/catch to handle the resolved (rather than rejected) prestart check promise value. Network errors are logged immediately in the `apiRequest` configuration, and this gate is primarily used to determine whether the form should be entered, or if the user should be blocked from entering the form. Feedback on the new approach appreciated.